data-contracts/README.md

```diff
@@ -0,0 +1,130 @@
+* Using the newly created Schema Registry, data engineers and architects define the schema of the events that comprise the organization's canonical data model - i.e. entities, events, and commands that are shared across applications. - along with other parts of the data contract. This includes data quality rules, metadata, and migration rules. A gradle plugin is utilized to register the schemas and related elements of the data contract with the Schema Registry.
+* Applications which producer and/or consume these event types can download the schemas from the Schema Registry. In our example, this is a JVM application built using Gradle. A gradle plugin is used to download the schemas, after which another gradle plugin is used to generate Java classes from those schemas - thus providing the application with compile-time type safety.
+
+### Prerequisites
+
+Clone the `confluentinc/demo-scene` GitHub repository (if you haven't already) and navigate to the `demo-scene` directory:
+
+```shell
+git clone git@github.com:confluentinc/demo-scene.git
+cd demo-scene
+```
+
+Here are the tools needed to run this tutorial:
+* [Confluent Cloud](http://confluent.cloud)
+* [Confluent CLI](https://docs.confluent.io/confluent-cli/current/install.html)
+* [Terraform](https://developer.hashicorp.com/terraform/install?product_intent=terraform)
+* [jq](https://jqlang.github.io/jq/)
+* [Gradle](https://gradle.org/install/)
+* JDK 17
+* IDE of choice
+
+> [!Note] When installing and configuring the Confluent CLI, include the Confluent Cloud credentials as environment variables for future use. For instance with bash or zsh, include these export statements:
+> 
+> ```shell
+> export CONFLUENT_CLOUD_API_KEY=<API KEY>
+> export CONFLUENT_CLOUD_API_SECRET<API SECRET>
+> ```
+>
+
+Terraform can use the value of any environment variable whose name begins with `TF_VAR_` as the value of a terraform variable of the same name. For more on this functionality, see the [terraform documentation](https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_var_name).
+
+Our example requires we set the value the `org_id` variable from Confluent Cloud. This denotes which organization will house the Confluent Cloud assets we are creating. So let's export the Confluent Cloud organization ID to a terraform environment variable.
+
+This command may open a browser window asking you to authenticate to Confluent Cloud. Once that's complete, the result of
+`confluent organization list` is queried by `jq` to extract the `id` of the current organization to which you are authenticated:
+
+```shell
+export TF_VAR_org_id=$(confluent organization list -o json | jq -c -r '.[] | select(.is_current)' | jq '.id')
+```
+
+### Create Assets in Confluent Cloud
+
+From the root of `data-contracts`, run the provided setup script - `setup.sh`. Note, you may need to edit permissions to make this file executable.
```

Comment:
suggest giving a heads up that the script will take a few mins to complete. either here, or below where it says "Let's have a look ..."